InformationScripting/src/queries/RuntimeQuery.h

```diff
@@ -41,7 +43,9 @@ class INFORMATIONSCRIPTING_API RuntimeQuery : public LinearQuery
 	private:
 		friend class QueryRegistry;
 
-		RuntimeQuery(Model::Node* target, QStringList);
+		QueryExecutor* executor_{};
+
+		RuntimeQuery(Model::Node* target, QStringList, QueryExecutor* executor = nullptr);
```

Comment:
Well you can't use partial specialization on methods, so things would need another indirection.